catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/security/WorkspacePolicyExtension.java

```diff
@@ -36,105 +33,102 @@
-            KeyValueCollectionPermission collection) {
+    private List<KeyValuePermission> getPermissions(KeyValueCollectionPermission collection) {
         return collection.getPermissionList()
                 .stream()
                 .filter(p -> p instanceof KeyValuePermission)
                 .map(p -> (KeyValuePermission) p)
                 .collect(Collectors.toList());
     }
 
-    private Stream<KeyValuePermission> overrides() {
-        return Stream.of(new KeyValuePermission(systemUserAttribute, ImmutableSet.of(
-                systemUserAttributeValue)));
+    private Map<String, Set<String>> groupPermissionsByKey(List<KeyValuePermission> permissions) {
+        return permissions.stream()
+                .collect(Collectors.toMap(KeyValuePermission::getKey,
+                        KeyValuePermission::getValues,
+                        Sets::union));
     }
 
-    @Override
-    public KeyValueCollectionPermission isPermittedMatchAll(CollectionPermission subject,
-            KeyValueCollectionPermission match) {
-
-        List<KeyValuePermission> permissions = getPermissions(match);
-
-        Stream<KeyValuePermission> remapped = permissions.stream()
-                .filter(byKeys(keyMapping.keySet()))
-                .map(remapKeys(keyMapping));
-
-        boolean matched = concat(overrides(), remapped).filter(subject::implies)
-                .findFirst()
-                .isPresent();
-
-        if (!matched) {
-            return match;
-        }
-
-        if (isOwnerPermission(getPermissions(match), subject)) {
-            return new KeyValueCollectionPermission();
-        }
-
-        return new KeyValueCollectionPermission(match.getAction(),
-                permissions.stream()
-                        .filter(byKeys(keyMapping.keySet()).negate())
-                        .collect(Collectors.toList()));
+    private Predicate<CollectionPermission> systemPred() {
+        Set<String> values = ImmutableSet.of(config.getSystemUserAttributeValue());
+        KeyValuePermission perm = new KeyValuePermission(config.getSystemUserAttribute(), values);
+        return (subject) -> subject.implies(perm);
     }
 
-    private boolean isOwnerPermission(List<KeyValuePermission> permissions,
-            CollectionPermission subject) {
-        Optional<KeyValuePermission> ownerPerm = permissions.stream()
-                .filter(keyValuePermission -> keyValuePermission.getKey()
-                        .equals(Core.METACARD_OWNER))
-                .findAny();
-
-        if (ownerPerm.isPresent()) {
-            KeyValuePermission ownerEmailPermission =
-                    new KeyValuePermission(Constants.EMAIL_ADDRESS_CLAIM_URI,
-                            ownerPerm.get()
-                                    .getValues());
-            return subject.implies(ownerEmailPermission);
+    private Predicate<CollectionPermission> pred(Map<String, Set<String>> p, String k1, String k2) {
```

Comment:
❗️ These variables need to be named something so they give insight. (I had to dig to realize that `p` is representing `permissions`, so it should be named as such).

Reply:
I will rename it to permissions then.